DEVELOPMENT.md

```diff
@@ -1,24 +1,117 @@
+  can be merged directly to master or via a feature branch. Furthermore, our ability to roll out upgrades to the live network via HCUs will significantly increase over time.     
+* For most **feature branches**, the protocol changes they implement will hopefully be small enough to be deployed via an HCU (and hence can be merged to `master`).
+  Though, for the foreseeable future, large and broadly-breaking improvements will be necessary for a fast evolution and feature-completion of Flow.
+  Such features should be kept in their individual feature branches.  
+* The `spork` branch, will be created in preparation of a new spork from the most recent `master`. Feature branches holding HCU-**in**compatible upgrades will be
+  consolidated on this `spork` branch. The spork branch is intended to be very short-lived and only exist for a few weeks right before the testnet spork.
+  When the Flow network has committed to the spork on a governance level (at the moment, largely coordinated by the Flow Foundation), the `spork` branch will be
+  merged back to `master`.
+
+We are purposefully continuing the usage of the 'spork' terminology here, to describe a "severely breaking change".
+Specifically, spork describes a deployment of new node software, which requires the entire network to be stopped and rebooted.
+We only carry over a snapshot of the execution state but fully re-initialize the protocol state. Thereby, major upgrades of the protocol,
+incl. migrations of the entire execution state to a new formats become reasonably straight forward.
+Additional context on HCU vs sporks can be found [here](https://developers.flow.com/networks/node-ops/node-operation/hcu#hcu-versus-spork).   
+
+<!---
+TEMPORARY to show the image, will be replaced with commented-out version below before merge 
+-->
+
+<img src='https://github.com/onflow/flow-go/blob/ff8882b2fcef5b7dfa86ac97cde407cc7b9ae3d5/docs/flow-go_branching-convention.png' width='350'>
+
+<!---
+<img src='https://github.com/onflow/flow-go/blob/master/docs/BranchingConvention.png' width='350'>
+-->
+
+
+
+#### Motivations
+* Based on the decentralized nature of blockchain networks, continuous deployment is practically intractable (without compromising decentralization).
+Therefore, longer wait times for features until their deployment must be accounted for in the development process. This convention has been incrementally
+developed and refined based on the multi-year learnings of managing the deployment of the flow network.
+* Deployment branches, `master` containing a runnable snapshot of the most recent development state and feature branches are intuitively exactly what you
+  would expect without further flow-specific context. We just _extend_ the common convention by a few edge cases accounting for the upgrade-constraints of a blockchain network:
+  * On a specific deployment branch, there can only be non-braking changes. This is to prevent accidents where we roll out new node
```

Comment:
```suggestion
  * On a specific deployment branch, there can only be non-breaking changes. This is to prevent accidents where we roll out new node
```